libyul/backends/wasm/WasmDialect.cpp

```diff
@@ -85,6 +87,69 @@ WasmDialect const& WasmDialect::instance()
 	return *dialect;
 }
 
+void WasmDialect::addEthereumExternals()
+{
+	// These are not YulStrings because that would be too complicated with regards
+	// to the YulStringRepository reset.
+	static string const i64{"i64"};
+	static string const i32{"i32"};
+	static string const i32ptr{"i32"}; // Uses "i32" on purpose.
+	struct External { string name; vector<string> parameters; vector<string> returns; };
+	static vector<External> externals{
+		{"getAddress", {i32ptr}, {}},
+		{"getExternalBalance", {i32ptr, i32ptr}, {}},
+		{"getBlockHash", {i64, i32ptr}, {i32}},
+		{"call", {i64, i32ptr, i32ptr, i32ptr, i32}, {i32}},
+		{"callDataCopy", {i32ptr, i32, i32}, {}},
+		{"getCallDataSize", {}, {i32}},
+		{"callCode", {i64, i32ptr, i32ptr, i32ptr, i32}, {i32}},
+		{"callDelegate", {i64, i32ptr, i32ptr, i32}, {i32}},
+		{"callStatic", {i64, i32ptr, i32ptr, i32}, {i32}},
+		{"storageStore", {i32ptr, i32ptr}, {}},
+		{"storageLoad", {i32ptr, i32ptr}, {}},
+		{"getCaller", {i32ptr}, {}},
+		{"getCallValue", {i32ptr}, {}},
+		{"codeCopy", {i32ptr, i32, i32}, {}},
+		{"getCodeSize", {i32ptr}, {}},
+		{"getBlockCoinbase", {i32ptr}, {}},
+		{"create", {i32ptr, i32ptr, i32, i32ptr}, {i32}},
+		{"getBlockDifficulty", {i32ptr}, {}},
+		{"externalCodeCopy", {i32ptr, i32ptr, i32, i32}, {}},
+		{"getExternalCodeSize", {i32ptr}, {i32}},
+		{"getGasLeft", {}, {i64}},
```

Comment:
Is the return going to stay 64bits for gas left?

Reply:
That is how it is defined for now.